core/services/ocr2/delegate.go

```diff
@@ -938,13 +938,19 @@ func (d *Delegate) newServicesOCR2Keepers(
 		d.cfg.JobPipeline().MaxSuccessfulRuns(),
 	)
 
+	var chain evm.Chain
```

Comment:
this chain will be just nil
you can reference `EVMDependencies20` how it gets a proper `chain` for this service and replicate the logic here.

Reply:
func EVMDependencies20(spec job.Job, db *sqlx.DB, lggr logger.Logger, set evm.ChainSet, pr pipeline.Runner) (evmrelay.OCR2KeeperProvider, *kevm20.EvmRegistry, Encoder, *kevm20.LogProvider, error) {
	var err error
	var chain evm.Chain
	var keeperProvider evmrelay.OCR2KeeperProvider
	var registry *kevm20.EvmRegistry

	oSpec := spec.OCR2OracleSpec

	// get the chain from the config
	chainID, err2 := spec.OCR2OracleSpec.RelayConfig.EVMChainID()
	if err2 != nil {
		return nil, nil, nil, nil, err2
	}
	chain, err2 = set.Get(big.NewInt(chainID))
	if err2 != nil {
		return nil, nil, nil, nil, fmt.Errorf("%w: %s", ErrNoChainFromSpec, err2)
	}